src/zenml/cli/stack_components.py

```diff
@@ -145,21 +202,43 @@ def generate_stack_component_list_command(
 
     def list_stack_components_command() -> None:
         """Prints a table of stack components."""
+
+        cli_utils.print_active_profile()
+        cli_utils.print_active_stack()
+
         repo = Repository()
+
         components = repo.get_stack_components(component_type)
         display_name = _component_display_name(component_type, plural=True)
         if len(components) == 0:
             cli_utils.warning(f"No {display_name} registered.")
             return
-        try:
-            active_component_name = repo.active_stack.components[
-                component_type
-            ].name
-        except KeyError:
-            active_component_name = None
+
+        local_stack = repo.active_stack_name
+        global_stack = repo.active_profile.active_stack or ""
+
+        global_active_component_name = None
```

Comment:
Referring to my other comment about what to show the user. showing the user both the globally active and the locally active version of everything introduces so much extra code overhead in many places. if we instead just show the local information, except when either currently outside of a repo or supplying a --global flag, I feel like this would simplify a lot of implementations, and doesn't deprive the user of any valuable information (if they reallly need to know both, they just issue the command twice).

Reply:
I completely agree and now that I've gone through the exercise of removing the global/local distinction wherever it didn't make sense anymore, I'm flabbergasted by the amount of useless code that I took out.